File: ProgrammingSkills/SwapNodesInPairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        
        aux = ListNode(head.next.val, ListNode(head.val, None))
        
        aux.next.next = self.swapPairs(head.next.next)
        return aux

File: JSProblems/SwapNodesInPairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    if(head == null || head.next == null){
        return head;
    }
    let aux = new ListNode(0);
    aux.next = head;
    let current = aux;
    while(current.next != null && current.next.val != null){        
            let local1 = current.next;
            let local2 = current.next.next
            current.next = local2;
            local1.next = local2.next;
            local2.next = local1;
            current = current.next.next;        
        }    
    return aux.next; 
};
